packages/core/src/Stores/Modules/Cashier/cashier-store.js

```diff
@@ -201,6 +201,29 @@ export default class CashierStore extends BaseStore {
         this.onSwitchAccount(this.accountSwitcherListener);
     }
 
+    // Initial init just for displaying P2P banner without mounting the entire cashier
+    @action.bound
+    init() {
+        // eslint-disable-next-line no-undef
+        reaction(
+            () => [this.root_store.client.is_logged_in, this.root_store.client.residence],
+            () => {
+                if (!this.is_p2p_visible && !this.root_store.client.is_virtual) {
+                    WS.authorized.p2pAdvertiserInfo().then(advertiser_info => {
+                        const advertiser_error = getPropertyValue(advertiser_info, ['error', 'code']);
+                        if (advertiser_error === 'RestrictedCountry') {
+                            this.setIsP2PBannerVisible(false);
+                        } else {
+                            this.setIsP2PBannerVisible(true);
+                        }
```

Comment:
IMO writing it as `this.setIsP2PBannerVisible(advertiser_error !== 'RestrictedCountry')` is clearer, but I understand you were following the other implementation in onmount common, so up to you

Reply:
Yep, agreed that it makes it neater. Will implement it.